crates/ruff/src/rules/ruff/rules/ambiguous_unicode_character.rs

```diff
@@ -103,50 +101,133 @@ pub(crate) fn ambiguous_unicode_character(
 
     let text = locator.slice(range);
 
+    // Most of the time, we don't need to check for ambiguous unicode characters at all.
+    if text.is_ascii() {
+        return diagnostics;
+    }
+
+    // Iterate over the "words" in the text.
+    let mut flags = WordFlags::empty();
+    let mut buffer = vec![];
```

Comment:
Can we use a more specific word for `buffer`? I struggle understanding the logic because the name doesn't tell me anything about what data gets stored in the buffer or its intended lifetime.

E.g. word_diagnostics